org.eclipse.corrosion/src/org/eclipse/corrosion/CorrosionPreferencePage.java

```diff
@@ -382,34 +382,43 @@ private void createCommandPathsPart(Composite container) {
 	private void installCommands() {
 		installButton.setText(Messages.CorrosionPreferencePage_installing);
 		installButton.setEnabled(false);
-		Job.create(Messages.CorrosionPreferencePage_installingRustupCargo, (ICoreRunnable) monitor -> {
-			try {
-				Bundle bundle = CorrosionPlugin.getDefault().getBundle();
-				URL fileURL = FileLocator.toFileURL(bundle.getEntry("scripts/rustup-init.sh")); //$NON-NLS-1$
-				File file = new File(new URI(fileURL.getProtocol(), fileURL.getPath(), null));
-				file.setExecutable(true);
-				String[] command = new String[] { "/bin/bash", "-c", file.getAbsolutePath() + " -y" }; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
-				ProcessBuilder builder = new ProcessBuilder(command);
-				Process process = builder.start();
-				if (process.waitFor() == 0) {
+
+		String[] command;
+		try {
+			Bundle bundle = CorrosionPlugin.getDefault().getBundle();
+			URL fileURL = FileLocator.toFileURL(bundle.getEntry("scripts/rustup-init.sh")); //$NON-NLS-1$
+			File file = new File(new URI(fileURL.getProtocol(), fileURL.getPath(), null));
+			file.setExecutable(true);
+			command = new String[] { "/bin/bash", "-c", file.getAbsolutePath() + " -y" }; //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$
+		} catch (IOException | URISyntaxException e) {
+			CorrosionPlugin.showError(Messages.CorrosionPreferencePage_cannotInstallRustupCargo,
+					Messages.CorrosionPreferencePage_cannotInstallRustupCargo_details, e);
+			return;
+		}
+
+		CommandJob installCommandJob = new CommandJob(command, Messages.CorrosionPreferencePage_installingRustupCargo,
+				Messages.CorrosionPreferencePage_cannotInstallRustupCargo,
+				Messages.CorrosionPreferencePage_cannotInstallRustupCargo_details, 15);
+		installCommandJob.setUser(true);
```

Comment:
@mickaelistria, I have set the installation command job with `setUser(true)` but do not get the blocking progress dialog, just in the progress tab. Does it work for you?

Reply:
OK, I guess the behavior of join depends on the preference may depend on the "Always run in background property". If you do want to show a dialog anyway, maybe you can try `workbench.getProgressService().showInDialog(getShell(), installCommandJob);`

Reply:
Regardless of my "Always run in background" property join and showInDialog do not work. Only thing that was successful at getting a dialog was creating a ProgressMonitorDialog, which seems to be no longer proper technique judging by this [FAQ](https://wiki.eclipse.org/FAQ_How_do_I_switch_from_using_a_Progress_dialog_to_the_Progress_view%3F).

After getting the dialog up, I did realize that blocking the UI for this long running event wasn't an enjoyable UX  and now am in even more favour of the progress view so with your +1 I'd merge this.